repository.py

```diff
@@ -130,14 +130,15 @@ def _parse_repodata(self, accessor):
         repomdfp.close()
 
         primaryfp = accessor.openAddress(primary_location)
-        # Open compressed xml using cpiofile._Stream which is an adapter between CpioFile and a stream-like object.
-        # Useful when specifying the URL for HTTP or FTP repository - A simple GzipFile object will not work in this situation.
-        primary_xml = cpiofile._Stream("", "r", "gz", primaryfp, 20*512)
+        # HTTP/FTP accessors don't implement tell(), so read the data into a buffer
+        primarydata = BytesIO(primaryfp.read())
+        primary_xml = gzip.GzipFile("", "r", 9, primarydata)
```

Comment:
parse() takes a filename or file-like object. Why not pass the BytesIO directly to parse()?

Reply:
I suppose for the missing `tell()` method. Or you mean the minidom `parse` function? In this case you need not compressed data

Reply:
Sorry, I got confused seeing the `9` compression level and thought it was compressing some uncompressed data.

I think it would be less confusing to call it like `gzip.GzipFile("", "r", fileobj=primarydata)` but that is a minor complaint.